### PR TITLE
set a default timeout of 10s for call()s

### PR DESCRIPTION
**Pull request by avahowell**

### CODE REVIEW

**Comment by DavidVorick** [comment]
We had an offline discussion about how some more error handling needs to be added, leaving this open until that's in place.

**Comment by avahowell** [comment]
the default timeout has been upped to 10 seconds. The error handling changes will be made in `Sia-UI`, not this repo.

File: src/sia.js [hunk 1]
```diff
@@ -27,6 +27,9 @@ const call = (address, opts) => new Promise((resolve, reject) => {
 	}
 	callOptions.url = 'http://' + address + callOptions.url
 	callOptions.json = true
+	if (typeof callOptions.timeout === 'undefined') {
+		callOptions.timeout = 10000
+	}
```
Comment (lukechampine) [inline src/sia.js:32]: this is just slightly-safer longhand for `callOptions.timeout = callOptions.timeout || 10000`, right?

Comment (avahowell) [reply src/sia.js:32]: Yes, though I think they are semantically equivalent. Writing it out longhand removes the implicit `undefined is false` type coercion

**Comment by VoidingWarranties** [comment]
@lukechampine wouldn't that prevent you from manually setting timeout of 0?

**Comment by lukechampine** [comment]
perhaps -- it depends on how `callOptions.timeout` is processed, I guess. If it interprets 0 as no timeout, then you can pass 0 and it'll work fine. But if it takes the same approach as we do here (only `undefined` means no timeout), then it's no longer possible.

**Comment by DavidVorick** [comment]
will this cause problems for anyone expecting the timeout to be unlimited for `/renter/download` calls?

**Comment by avahowell** [comment]
Just to update some of the concerns in this PR, setting a `timeout: 0` results in an unlimited timeout. Should be ready to merge.

**Pull request merged**